doc/specs/#2697 - PowerShell DSC resources.md

```diff
@@ -0,0 +1,256 @@
+This could live inside the DSC resource, but in order to keep the code contained and easy to manage and test is better to expose it.
+
+#### DSC Resource
+
+```
+WinGetUserSettings [String] #ResourceName
+{
+    Settings = [string[]]
+    SID = [string]
+    [Action = [string]]
+    [DependsOn = [string[]]]
+    [PsDscRunAsCredential = [PSCredential]]
+}
+```
+
+*Settings*
+
+A Hashtable with the user settings to be set.
+
+*SID (Key)*
+
+DSC resources require a key that is a string, a signed/unsigned integer, or Enum types. The key must be unique per resource, but we only expect to have one winget user settings resource in a session. To resolve this, we can use the SID of the current user as the key. This will also allow a future implementation where an administrator sets different user settings for different users. For now, we won't expect people to set this, we will do it internally.
+
+*Action*
+
+Action is an enum with two values: Partial, Full.
+- Partial: For Set, this means the changes are additive and overwrite existing settings if already present. For Test, this means validate all the settings in the input Settings are present and have the same value.
+- Full: For Set, this means overwrite the settings.json file with the input settings. For Test, this means validate all the settings in the input Settings are present and have the same value and no other settings are explicitly set.
+
+The default value is Partial.
+
+#### Example of usage
+```
+$resource = @{
+    Name = 'WinGetUserSettings'
+    ModuleName = 'Microsoft.WinGet.DSC'
+    Property = @{
+        Settings = @{
+            telemetry = @{
+                disable = $true
+            }
+        }
+    }
+}
+
+$testResult = Invoke-DscResource @resource -Method Test
+if (-not $testResult.InDesiredState)
+{
+    Invoke-DscResource @resource -Method Set
+}
+
+```
+
+### WinGetAdminSettings
+This resource will be in charge of enabling and disabling winget administrator settings and is a wrapper around Enable-WinGetSetting and Disable-WinGetSetting.
+
+#### winget-cli
+Currently, there's no way to expose if the administrator settings are enabled or not. This is a requirement for both Test and Get. We need to expose this information with a new `--export` option in the `winget settings` command. The export option will print a json with the information required.
+
+```
+{
+    "LocalManifestFiles": true,
+    "BypassCertificatePinningForMicrosoftStore": false,
+    "FutureAdminSetting": true
+}
+```
+We could technically read the registry and avoid this, but that doesn't sound like the right thing to do.
+
+#### DSC Resource
+
+```
+WinGetAdminSettings [String] #ResourceName
+{
+    Settings = [string[]]
+    SID = [string]
+    [DependsOn = [string[]]]
+    [PsDscRunAsCredential = [PSCredential]]
+}
+```
+
+*Settings*
+
+A Hashtable with the admin settings to be set.
```

Comment:
Is this to future proof against non-boolean admin settings?

Reply:
I was thinking it as more of a "don't update the module every time there's a new admin setting"